plugins/micromega/coq_micromega.ml

```diff
@@ -1978,13 +2015,38 @@ type provername = string * int option
 
 open Persistent_cache
 
-module Cache = PHashtable(struct
-  type t = (provername * micromega_polys)
-  let equal = (=)
-  let hash  = Hashtbl.hash
-end)
 
-let csdp_cache = ".csdp.cache"
+module MakeCache(T : sig type prover_option
+                         type coeff
+                         val  hash_coeff : int -> coeff -> int
+                 end) =
+  struct
+    module E =
+      struct
+        type t = T.prover_option * (T.coeff Mc.pol * Mc.op1) list
+
+        let equal = (=)
```

Comment:
Can you write a dedicated equality function here? Parametric equality is known to introduce slowness, it's always better to write out your own equality cheks.